pandas/core/frame.py

```diff
@@ -5300,8 +5304,10 @@ def reorder_levels(self, order, axis=0) -> "DataFrame":
         result = self.copy()
 
         if axis == 0:
+            assert isinstance(result.index, ABCMultiIndex)
```

Comment:
isn't this is a breaking change?

this PR
```python
>>> pd.DataFrame().swaplevel()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\pandas\pandas\core\frame.py", line 5270, in swaplevel
    assert isinstance(result.index, ABCMultiIndex)
AssertionError
>>>
```

on master
```python
>>> pd.DataFrame().swaplevel()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\pandas\pandas\core\frame.py", line 5268, in swaplevel
    result.index = result.index.swaplevel(i, j)
AttributeError: 'Index' object has no attribute 'swaplevel'
>>>
```

Reply:
Yeah, I've made it a TypeError, to mirror reorder_levels. I also added a Whatsnew note.

Reply:
BTW, is this an API change? We're reaching 1.0, so in principle we should be wary on changing the API. Should I raise an AttributeError instead of TypeError to maintain compatibility (though that is strictly a wrong error to raise)?